### PR TITLE
ICU-22309 update CONTRIBUTING.md for new process

### PR DESCRIPTION
**Pull request by srl295**
##### Checklist

- [x] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-22309
- [x] Required: The PR title must be prefixed with a JIRA Issue number. 
- [x] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [x] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [x] Issue accepted (done by Technical Committee after discussion)

#2402 was the prior update to CONTRIBUTING.md
#2415 also see

CLA section from https://github.com/unicode-org/.github/blob/main/.github/CONTRIBUTING.md

fyi @annebright

### CODE REVIEW

**Comment by srl295** [comment]
@markusicu ping ? required for new CLA process

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -2,35 +2,37 @@
...
 
- (As of 2018-Aug, the CLA assistant is not yet set up to recognize an organization affiliation, so individuals still have to sign it for now.)
+SPDX-License-Identifier: Unicode-DFS-2016
 
 ## Process
 
 See also [git for ICU Developers][git4icu].
 
-If you want to join the team, then please contact us. Once everything is agreed, the ICU team adds you to the GitHub project and the Jira issue tracker.
+If you want to join the team, then please [contact us][contacts]. Once everything is agreed, the ICU team adds you to the [GitHub project][unicode-org/icu] and the [Jira issue tracker][bugs].
```
Comment (markusicu) [inline CONTRIBUTING.md:27]: It looks like casual patches (without joining the team) got lost in the shuffle. I suggest we bring it back roughly here:

```suggestion
For patches, please submit both a Jira ticket and a GitHub pull request. See [Submitting ICU Bugs and Feature Requests][bugs].

If you want to join the team, then please [contact us][contacts]. Once everything is agreed, the ICU team adds you to the [GitHub project][unicode-org/icu] and the [Jira issue tracker][bugs].
```

Comment (srl295) [reply CONTRIBUTING.md:27]: @annebright @skywalkerswartz ^

Comment (srl295) [reply CONTRIBUTING.md:27]: @markusicu could we consider this in a separate PR? So that the wording for the already-installed CLA is in place first?

Comment (markusicu) [reply CONTRIBUTING.md:27]: It seems like it could have been quick to amend the commit... but if you can remember to make another PR then that's ok.

**Review by markusicu — APPROVED** [review]

**Pull request merged**